src/overlays/actors/ovl_Bg_Heavy_Block/z_bg_heavy_block.c

```diff
@@ -21,35 +38,497 @@ const ActorInit Bg_Heavy_Block_InitVars = {
+}
+
+void BgHeavyBlock_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHeavyBlock* this = THIS;
+    switch (this->dyna.actor.params & 0xFF) {
+        case 2:
+            break;
+        case 3:
+            break;
+        default:
+            DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void BgHeavyBlock_MovePiece(BgHeavyBlock* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+
+    thisx->velocity.y += thisx->gravity;
+
+    if (thisx->velocity.y < thisx->minVelocityY) {
+        thisx->velocity.y = thisx->minVelocityY;
+    }
+
+    thisx->velocity.x *= 0.98f;
+    thisx->velocity.z *= 0.98f;
+    func_8002D7EC(thisx);
+    thisx->shape.rot.x += thisx->posRot.rot.x;
+    thisx->shape.rot.y += thisx->posRot.rot.y;
+    thisx->shape.rot.z += thisx->posRot.rot.z;
+
+    if (!(this->pieceFlags & HEAVYBLOCK_HIT_FLOOR)) {
+        thisx->posRot.pos.y += this->unk_164.y;
+        thisx->pos4.y += this->unk_164.y;
+        func_8002E4B4(globalCtx, thisx, 50.0f, 50.0f, 0.0f, 5);
+        thisx->posRot.pos.y -= this->unk_164.y;
+        thisx->pos4.y -= this->unk_164.y;
+        if (thisx->bgCheckFlags & 1) {
+            this->pieceFlags |= HEAVYBLOCK_HIT_FLOOR;
+            thisx->velocity.y = Math_Rand_ZeroFloat(4.0f) + 2.0f;
+            thisx->velocity.x = Math_Rand_CenteredFloat(8.0f);
+            thisx->velocity.z = Math_Rand_CenteredFloat(8.0f);
+            BgHeavyBlock_SetPieceRandRot(this, 1.0f);
+            Audio_PlayActorSound2(thisx, NA_SE_EV_ROCK_BROKEN);
+            func_800AA000(thisx->xzDistFromLink, 0x96, 0xA, 8);
+        }
+    }
+
+    if (this->timer > 0) {
+        this->timer -= 1;
+    } else {
+        Actor_Kill(thisx);
+    }
+}
+
+void BgHeavyBlock_SpawnDust(GlobalContext* globalCtx, f32 posX, f32 posY, f32 posZ, f32 velX, f32 velY, f32 velZ,
+                            u8 dustParams) {
+    Color_RGBA8_n primColor;
+    Color_RGBA8_n envColor;
+    Vec3f eye;
+    Vec3f at;
+    s16 sp6E;
+    s16 sp6C;
+    Vec3f accel;
+    Vec3f velocity;
+    Vec3f pos;
+    f32 sp44;
+    s16 scaleStep;
+    s16 scale;
+
+    pos.x = posX;
+    pos.y = posY;
+    pos.z = posZ;
+
+    if ((dustParams & 1)) {
+        // red dust, landed in fire
+        primColor.r = 150;
+        primColor.g = primColor.b = envColor.g = envColor.b = 0;
+        envColor.r = 80;
+        primColor.a = envColor.a = 0;
+    } else {
+        // brown dust
+        // clang-format off
+        primColor.r = 170; primColor.g = 130; primColor.b = 90; primColor.a = 255;
+        envColor.r = 100; envColor.g = 60; envColor.b = 20; envColor.a = 255;
+        // clang-format on
+    }
+
+    accel.z = 0.0f;
+    accel.x = 0.0f;
+    accel.y = (dustParams & 8) ? 0.0f : 0.5f;
+
+    eye = globalCtx->cameraPtrs[globalCtx->activeCamera]->eye;
+    at = globalCtx->cameraPtrs[globalCtx->activeCamera]->at;
```

Comment:
`ACTIVE_CAM` should be used here, ideally with a temp to avoid repeating it ofc, tho I assume that's not possible here.
```suggestion
    eye = ACTIVE_CAM->eye;
    at = ACTIVE_CAM->at;
```

Reply:
ye you gave me shit for it in my stream so I thought you actually wanted me to take it out lol. a temp doesnt work

Reply:
Ah yeah I wasn't saying anything about using the macro, I meant the part about repeating it :P